chall04/edramire.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python3
+
+import sys
+import fileinput
+
+
+def process_file(filename):
+    shelfs = []
+    books = 0
+    try:
+        for line in fileinput.input(filename):
+            shelfs_str = line.split(" ")
+            if len(shelfs) == 0:
+                for word in shelfs_str:
+                    word = word.rstrip('\n\r')
+                    if len(word) == 0:
+                        continue
+                    if not word.isdigit():
+                        print(f"./edramire.py: {filename or 'stdin'}: Can't read file")
```

Comment:
If `filename` is an empty string and is not `None`, you will try to read stdin because `"" or "stdin"` will always return `"stdin"`. However if filename isn't `None` it means that you somehow tried to pass `""` as a file, and you should not try to read stdin. In some cases, such a behavior could cause various problems.

It is not an error I would penalize here, but if you run `./edramire.py ""` it will in fact pass a parameter `""` (being an empty string), so you could have simulated this error case.